USF 2382 - In cart flavor & size editing (#383)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/cart/tutorials/enable-product-variation-updates-in-cart.mdx

// Later in the code, inside the Footer slot
if (ctx.item?.itemType === 'ConfigurableCartItem' && enableUpdatingProduct === 'true') {
  const editLink = document.createElement('div');
  editLink.className = 'cart-item-edit-link';

  UI.render(Button, {
    children: placeholders?.Global?.CartEditButton,
    variant: 'tertiary',
    size: 'medium',
    icon: h(Icon, { source: 'Edit' }),
    onClick: () => handleEditButtonClick(ctx.item),
  })(editLink);

  ctx.appendChild(editLink);
}
```

> When a shopper clicks the **Edit** button, a modal opens with a mini-PDP interface that allows them to modify their product options. Upon successful update, comprehensive success messaging appears with auto-dismissing notifications.

Reviewer: ```suggestion
When a shopper clicks the **Edit** button, a modal opens with a mini-PDP interface that allows them to modify their product options. An auto-dismissing notification appears after a successful update.
```

<Task>
### Configure the Cart Summary Block

> The cart block shows edit buttons *by default* when configurable products are present. If you want to disable it:

Reviewer: ```suggestion
The cart block shows **Edit** buttons *by default* when configurable products are present. If you want to disable it:
```

5. Save the changes.
6. Preview the changes by clicking the **Preview** button.
7. Publish the changes by clicking the **Publish** button.

<Aside type="note">

>   The mini cart does not show edit buttons by default and requires explicit configuration to enable this feature.

Reviewer: ```suggestion
By default, the mini cart does not show **Edit** buttons and needs to be explicitly enabled.
```

## Feature Behavior

When the **Edit** button is clicked, the following happens:

1. **Modal Interface**: A mini-PDP modal opens directly over the current page, maintaining user context.
2. **Pre-populated Options**: The modal displays the product with current selections already chosen.
3. **In-place Updates**: Changes are applied to the existing cart item.
4. **Comprehensive Messaging**: Success notifications appear in:
   - The main cart notification area (if present)
   - The mini-cart message system
   - Both locations simultaneously for consistent feedback
5. **Auto-dismissing Notifications**: Messages automatically disappear for better UX.

<Aside type="tip">

>   The modal approach ensures users never lose their shopping context while making product modifications, leading to a more seamless shopping experience.

Reviewer: ```suggestion
Using modals ensures users don't lose their shopping context when making product modifications.
```
